### PR TITLE
Feature/develop authentication verification UI

### PR DESCRIPTION
**Pull request by christian-barrios995**
Name: "Develop Authentication and Verification Pages UI"
About: 
- Design and implement UI pages for user authentication and email verification.
- Include pages for signing up, signing in, and verifying email.
- Implement user-friendly interfaces with appropriate form fields and error handling.
Labels: feature
notion link: [Develop Authentication and Verification Pages UI](https://www.notion.so/PRO-MVP-Develop-Authentication-and-Verification-Pages-UI-b9d423b500f345ac9e1a00327f401ad0?pvs=4)

Important things to check:
reusability and features

---
Feature Description
- Created authentication UI
- Created Verification UI

---
Steps to Reproduce the Feature

- Sign-in Page: localhost:3000/sign-in
- Sign-up page: localhost: 3000/sign-up
- Verify password: localhost: 3000/verify-email

---
Expected Behavior

- Page loads normally
- Page Layout shows responsiveness

---
Test Environment
Browser: [e.g. Chrome, Safarri, Firefox]
Test Scenarios:

- [ ] Did the Test Feature Pass?

- [ ] Did you check the features on different browsers?

### CODE REVIEW

File: frontend/src/containers/SignInPage/SignInPage.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+}
+
+const SignInPage: React.FC<SignInPageProps> = ( ) => {
+  const [userCredentials, setUserCredentials] = useState({
+    email: '',
+    password: '',
+  })
+
+  const [borderColorValidation, setBorderColorValidation] = useState(
```
Comment (christian-barrios995) [inline frontend/src/containers/SignInPage/SignInPage.tsx]: Ate Roan, how should the invalid input look like?

File: frontend/src/containers/SignInPage/SignInPage.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  function handlePassword(inputValue: string){
+    setUserCredentials((prevUserCredentials) => ({
+      ...prevUserCredentials,
+      password: inputValue
+    }));
+  }
+
+  function handleCredentials(){
```
Comment (christian-barrios995) [inline frontend/src/containers/SignInPage/SignInPage.tsx:37]: Ate Roan. What more error validation should be added in the sign in?

File: frontend/src/containers/SignInPage/SignInPage.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+          maxWidth: '600px',
+          '&:hover': { backgroundColor: '#d2522b' },
+          textTransform: 'inherit',
+          fontSize: '24px'
+        }}
+      >
+        Login
+      </Button>
+      <Button
+        variant="contained"
```
Comment (christian-barrios995) [inline frontend/src/containers/SignInPage/SignInPage.tsx]: Ate Roan. Should sign-in and sign-up with google be removed? I forgot what we've talked before and could not find the notion notes.

File: frontend/src/containers/SignUpPage/SignUpPage.tsx [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+      ...prevUserCredentials,
+      password2: inputValue
+    }));
+  }
+
+  function handleSignUp(){
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    const isValidEmail = emailRegex.test(userCredentials.email);
+    //const isPasswordValid what is the condition for the password to be valid?
+
```
Comment (christian-barrios995) [inline frontend/src/containers/SignUpPage/SignUpPage.tsx:53]: @sence1roandayne Ate Roan, what else should be added or removed?

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -45,6 +48,9 @@ const App: React.FC = () => {
           element={<TermsAndConditionsPage />}
         />
         <Route path="/thank-you" element={<ThankYouPage />} />
+        <Route path='/sign-in' element={<SignInPage/>}/>
+        <Route path='/Sign-up' element={<SignUpPage/>}/>
```
Comment (sence1roandayne) [inline frontend/src/App.tsx]: make the /Sign-up small caps

File: frontend/src/containers/SignInPage/SignInPage.tsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+                Sign-in
+            </Typography>
+            <TextField
+                onChange={(e) => handleEmail(e.target.value)}
+                type='email'
+                id='email'
+                label="Email address"
+                placeholder='Input your email'
+                sx={{
+                    backgroundColor: '#F3F3F3',
```
Comment (sence1roandayne) [inline frontend/src/containers/SignInPage/SignInPage.tsx]: Let's make this white nalang. Like lahat ng bg color ng textfield components ay white nalang. dark bg color is commonly used for defaults pala and white sa mga pwedeng iedit ng users.

File: frontend/src/containers/SignInPage/SignInPage.tsx (hunk 3 above)
Comment (sence1roandayne) [reply frontend/src/containers/SignInPage/SignInPage.tsx]: Yes please remove it for now. We will only use email-password sign in and sign up

File: frontend/src/containers/SignInPage/SignInPage.tsx (hunk 1 above)
Comment (sence1roandayne) [reply frontend/src/containers/SignInPage/SignInPage.tsx]: Let's use Snackbar component for all success and error messages

File: frontend/src/containers/SignUpPage/SignUpPage.tsx [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+                    },
+                    shrink: false
+                  }}
+            />
+            <TextField
+                onChange={(e) => handlePassword(e.target.value)}
+                type='password'
+                id='password'
+                label="Password"
+                placeholder='Input your email'
```
Comment (sence1roandayne) [inline frontend/src/containers/SignUpPage/SignUpPage.tsx]: Update placeholder

File: frontend/src/containers/SignUpPage/SignUpPage.tsx [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+                    },
+                    shrink: false
+                  }}
+            />
+             <TextField
+                onChange={(e) => handlePassword2(e.target.value)}
+                type='password'
+                id='paswword2'
+                label="Password"
+                placeholder='Input your email'
```
Comment (sence1roandayne) [inline frontend/src/containers/SignUpPage/SignUpPage.tsx]: Update placeholder

File: frontend/src/containers/SignUpPage/SignUpPage.tsx [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+                        fontWeight: '700'
+                    },
+                    shrink: false
+                  }}
+            />
+             <TextField
+                onChange={(e) => handlePassword2(e.target.value)}
+                type='password'
+                id='paswword2'
+                label="Password"
```
Comment (sence1roandayne) [inline frontend/src/containers/SignUpPage/SignUpPage.tsx]: Make this Confirm Password

File: frontend/src/containers/SignUpPage/SignUpPage.tsx [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+  }
+
+  function handlePassword2(inputValue: string){
+    setUserCredentials((prevUserCredentials) => ({
+      ...prevUserCredentials,
+      password2: inputValue
+    }));
+  }
+
+  function handleSignUp(){
```
Comment (sence1roandayne) [inline frontend/src/containers/SignUpPage/SignUpPage.tsx:48]: Please double check this function. I'm not getting any error messages. I think hindi kasi naka wrap sa else yung`navigate('/verify-email')`

File: frontend/src/containers/SignUpPage/SignUpPage.tsx [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    const isValidEmail = emailRegex.test(userCredentials.email);
+    //const isPasswordValid what is the condition for the password to be valid?
```
Comment (sence1roandayne) [inline frontend/src/containers/SignUpPage/SignUpPage.tsx]: Min. 8 nalang

File: frontend/src/containers/SignUpPage/SignUpPage.tsx (hunk 4 above)
Comment (sence1roandayne) [reply frontend/src/containers/SignUpPage/SignUpPage.tsx:53]: Also check if email and pw are present. Add "required" FE validation

File: frontend/src/containers/SignInPage/SignInPage.tsx [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+            fontWeight: '300',
+            fontStyle: 'italic',
+            marginBottom: '10px',
+            textAlign: 'start',
+          }}
+          
+        >
+          Forgot password?
+        </Link>
+        <Link
```
Comment (sence1roandayne) [inline frontend/src/containers/SignInPage/SignInPage.tsx]: For links,
```
import { Link as MuiLink } from '@mui/material'
import { Link as RouterLink } from 'react-router-dom'

return (
   <MuiLink
          component={RouterLink}
          to="/sign-up"
          underline="none"
          variant="body1"
          sx={{
            cursor: 'pointer',
            fontSize: '16px',
            color: '#767676',
            marginBottom: '10px',
            textAlign: 'start',
          }}
        >
          No account yet? Sign-up here
     </MuiLink>
)
```

**Review by sence1roandayne — CHANGES_REQUESTED** [review]

**Comment by sence1roandayne** [comment]
@christian-barrios995 Please pull latest develop

**Review by sence1roandayne — APPROVED** [review]
LGTM

**Pull request merged**